java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -0,0 +1,189 @@
+  private ConstraintManager constraintManager;
+  MethodBehavior methodBehavior;
+
+  BytecodeEGWalker(BehaviorCache behaviorCache){
+    this.behaviorCache = behaviorCache;
+  }
+
+  public MethodBehavior getMethodBehavior(Symbol.MethodSymbol symbol, SquidClassLoader classLoader) {
+    methodBehavior = behaviorCache.methodBehaviorForSymbol(symbol);
+    if(!methodBehavior.isComplete()) {
+      execute(symbol, classLoader);
+      methodBehavior.completed();
+    }
+    return methodBehavior;
+  }
+
+  private void execute(Symbol.MethodSymbol symbol, SquidClassLoader classLoader) {
+    explodedGraph = new ExplodedGraph();
+    constraintManager = new ConstraintManager();
+    programState = ProgramState.EMPTY_STATE;
+    workList = new LinkedList<>();
+    endOfExecutionPath = new LinkedHashSet<>();
+    steps = 0;
+    BytecodeCFGBuilder.BytecodeCFG bytecodeCFG = BytecodeCFGBuilder.buildCFG(symbol, classLoader);
+    bytecodeCFG.blocks.forEach(b -> System.out.println(b.printBlock()));
+    for (ProgramState startingState : startingStates(symbol, programState)) {
+      enqueue(new ProgramPoint(bytecodeCFG.entry()), startingState);
+    }
+    while (!workList.isEmpty()) {
+      steps++;
+      if (steps > MAX_STEPS) {
+        throw new IllegalStateException("Too many steps");
+      }
+      // LIFO:
+      setNode(workList.removeFirst());
+      if (programPosition.block.successors().isEmpty()) {
+        endOfExecutionPath.add(node);
+        continue;
+      }
+
+      if (programPosition.i < programPosition.block.elements().size()) {
+        // process block element
+        executeInstruction((BytecodeCFGBuilder.Instruction) programPosition.block.elements().get(programPosition.i));
+      } else {
+        // process block exit, which is unconditional jump such as goto-statement or return-statement
+        handleBlockExit(programPosition);
+      }
+    }
+
+    handleEndOfExecutionPath();
+    executeCheckEndOfExecution();
+    // Cleanup:
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
+    System.out.println("instruc "+instruction.opcode+"  "+ Printer.OPCODES[instruction.opcode]);
+    // TODO execute check prestatement
+    switch (instruction.opcode) {
+      case Opcodes.ACONST_NULL:
+        programState = programState.stackValue(SymbolicValue.NULL_LITERAL);
+        break;
+      case Opcodes.ALOAD:
+      case Opcodes.DLOAD:
+      case Opcodes.FLOAD:
+      case Opcodes.ILOAD:
+        break;
+      case Opcodes.LLOAD:
+      case Opcodes.AALOAD:
+      case Opcodes.BALOAD:
+      case Opcodes.CALOAD:
+      case Opcodes.DALOAD:
+      case Opcodes.FALOAD:
+      case Opcodes.IALOAD:
+      case Opcodes.LALOAD:
+      case Opcodes.SALOAD:
+        break;
+      default:
+        // do nothing
+    }
+    // TODO execute check poststatement
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)